docs/modules/ROOT/pages/crosschain.adoc

```diff
@@ -0,0 +1,175 @@
++import "@openzeppelin/contracts-upgradeable/crosschain/CrossChainEnabled.sol";
+
+-contract MyToken is Initializable, ERC20Upgradeable, UUPSUpgradeable {
++abstract contract MyTokenCrossChain is Initializable, ERC20Upgradeable, UUPSUpgradeable, CrossChainEnabled {
+```
+
+Once that is done, we can use the `onlyCrossChainSender` modifier, provided by `CrossChainEnabled` in order to protect the minting and upgrading operations.
+
+```diff
+-    modifier onlyOwner() {
+-        require(owner == _msgSender(), "Not authorized");
+-        _;
+-    }
+
+-    function mint(address to, uint256 amount) public onlyOwner {
++    function mint(address to, uint256 amount) public onlyCrossChainSender(owner) {
+
+-    function _authorizeUpgrade(address newImplementation) internal override onlyOwner {
++    function _authorizeUpgrade(address newImplementation) internal override onlyCrossChainSender(owner) {
+```
+
+This change will effectively restrict the mint and upgrade operations to the `owner` on the remote chain.
+
+== Specializing for a specific chain
+
+Once the abstract cross-chain version of our token is ready we can easily specialize it for the chain we want, or more precisely for the bridge system that we want to rely on.
+
+This is done using one of the many `CrossChainEnabled` implementations.
+
+For example, if our token on xDai, and our governor on mainnet, we can use the https://docs.tokenbridge.net/amb-bridge/about-amb-bridge[AMB] bridge available on xDai at https://blockscout.com/xdai/mainnet/address/0x75Df5AF045d91108662D8080fD1FEFAd6aA0bb59[0x75Df5AF045d91108662D8080fD1FEFAd6aA0bb59]
+
+[source,solidity]
+----
+[...]
+
+import "@openzeppelin/contracts-upgradeable/crosschain/amb/CrossChainEnabledAMB.sol";
+
+contract MyTokenXDAI is
+    MyTokenCrossChain,
+    CrossChainEnabledAMB(0x75Df5AF045d91108662D8080fD1FEFAd6aA0bb59)
+{}
+----
+
+If token on mainnet, and our governor on optimism, we use the optimism https://community.optimism.io/docs/protocol/protocol-2.0/#l1crossdomainmessenger[CrossDomainMessenger] available on mainnet at https://etherscan.io/address/0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1[0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1]
+
+[source,solidity]
+----
+[...]
+
+import "@openzeppelin/contracts-upgradeable/crosschain/optimismCrossChainEnabledOptimism.sol";
+
+contract MyTokenOptimism is
+    MyTokenCrossChain,
+    CrossChainEnabledOptimism(0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1)
+{}
+----
+
+== Going further with access control
+
+In previous example, we didn't use the using xref:access-control.adoc#ownership-and-ownable[`Ownable`] pattern because it is not out of the box compatible with a cross-chain context that it was not designed for. Unlike xref:access-control.adoc#ownership-and-ownable[`Ownable`], xref:access-control.adoc#role-based-access-control[`AccessControl`] is more effective at capturing the nuances and can effectivelly be used to build cross-chain-aware contracts.
+
+[source,solidity]
+----
```

Comment:
Can we make this snippet simpler by only showing what changed? Or we can also first show the changes and then show the full code, so it can be used as a complete reference.